docs/user_manual/serialization.md

```diff
@@ -0,0 +1,159 @@
+
+- [`InhomogeneousComponentData`](#json-schema-inhomogeneous-component-data-object): `Object`
+  - [`Attribute`](#json-schema-attribute): [`AttributeValue`](#json-schema-attribute-value): the value of each attribute per attribute.
+
+#### JSON schema attribute value
+
+The [`AttributeValue`](#json-schema-attribute-value) contains the actual value of an attribute. The value can be any of `null` if it is not provided, or any of [`int32_t`](#json-schema-int32_t), [`int8_t`](#json-schema-int8_t), [`double`](#json-schema-double) or [`RealValueOutput`](#json-schema-realvalueoutput).
+The type is listed for each attribute in [Components](components.md).
+
+- [`AttributeValue`](#json-schema-attribute-value): `null` | [`int32_t`](#json-schema-int32_t) | [`int8_t`](#json-schema-int8_t) | [`double`](#json-schema-double) | [`RealValueOutput`](#json-schema-real-value)
+
+#### JSON schema int32_t
+
+An [`int32_t`](#json-schema-int32_t) is an integer `number` value usually representing an ID. It may be in the inclusive range $\left[-2^{31},+2^{31} - 1\right]$.
+The type is listed for each attribute in [Components](components.md).
+
+- [`int32_t`](#json-schema-int32_t): `number`
+
+#### JSON schema int8_t
+
+An [`int8_t`](#json-schema-int8_t) integer `number` value usually representing an enumeration value or a discrete setting. It may be in the inclusive range $\left[-2^{7},+2^{7} - 1\right]$.
+The type is listed for each attribute in [Components](components.md).
+
+- [`int8_t`](#json-schema-int8_t): `number`
+
+#### JSON schema double
+
+A [`double`](#json-schema-double) floating point `number` or `string` value usually representing a real.
+If it is a `string`, it shall be either `"inf"` or `"+inf"` for positive infinity, or `"-inf"` for negative infinity.
+Any other values are unsupported.
+The type is listed for each attribute in [Components](components.md).
+
+- [`double`](#json-schema-double): `number`|`string`
+
+#### JSON schema RealValueOutput
+
+A [`RealValueOutput`](#json-schema-realvalueoutput) of which the data format depends on the type of calculation.
+For symmetric calculations, it is a [`double`](#json-schema-double). For asymmetric calculations, it is an [`Array[double,3]`](#json-schema-double) of size 3.
+The type is listed for each attribute in [Components](components.md).
+
+- [`RealValueOutput`](#json-schema-realvalueoutput): `number` for symmetric calculations.
+- [`RealValueOutput`](#json-schema-realvalueoutput): `Array` of size 3 for asymmetric calculations, one for each phase.
+  - [`double`](#json-schema-double): the value for the `_a` phase.
+  - [`double`](#json-schema-double): the value for the `_b` phase.
+  - [`double`](#json-schema-double): the value for the `_c` phase.
+
+### msgpack serialization format specification
+
+The msgpack serialization format is a compressed version of the [JSON serialization format](#json-serialization-format-specification) and all features supported for JSON are also supported for msgpack.
```

Comment:
Please state here that for `msgpack`, `inf` and `-inf` will be directly represented in IEEE 754 `double` value, not via a string.